src/realm/object-store/property.hpp

```diff
@@ -127,6 +134,7 @@ struct Property {
 
     bool type_is_indexable() const noexcept;
     bool type_is_nullable() const noexcept;
+    bool type_is_nested() const noexcept;
```

Comment:
nesting_level would be better.

Reply:
what if 2 property have the same number of nesting levels but the different containers, eg. `List<List<Int>>` and `Dictionary<List<Int>>`, if we don't store the types and we use only the number of nested levels we could end up making an implicit conversion between these 2 containers (although things will crash once we'd try to insert something at storage level)

Reply:
what if 2 property have the same number of nesting levels but the different containers, eg. `List<List<Int>>` and `Dictionary<List<Int>>`, if we don't store the types and we use only the number of nested levels we could end up making an implicit conversion between these 2 containers (although things will crash once we'd try to insert something at storage level)

Reply:
fixed.